diff --git a/elements/avatar.js b/elements/avatar.js
index ffc5ba97fc393886b6420acb80b416338abad8d8..71670509c3c8ce121c810b58ded0d466c3686ec9 100644
--- a/elements/avatar.js
+++ b/elements/avatar.js
@@ -61,13 +61,13 @@ const style = css `
   :host([size='large']) {
     width: 3.8em;
   }
-  .slot:has-slotted {
+  .slot {
     place-items: center;
     place-content: center;
     border-color: ${theme.textColor};
   }
   .img,
-  .slot:has-slotted {
+  .slot {
     display: flex;
     width: 100%;
     aspect-ratio: 1;
diff --git a/elements/sort-box.js b/elements/sort-box.js
index 15a623e93a3366c2baf72a815dec34a0b9979923..e66e03d79efd6d6dc62c8e59c1883e0a19ed6879 100644
--- a/elements/sort-box.js
+++ b/elements/sort-box.js
@@ -130,7 +130,7 @@ let DuoyunSortItemElement = (() => {
     let _classDescriptor;
     let _classExtraInitializers = [];
     let _classThis;
-    let _classSuper = DuoyunGestureElement;
+    let _classSuper = GemElement;
     let _handle_decorators;
     let _handle_initializers = [];
     let _handle_extraInitializers = [];
diff --git a/elements/contextmenu.js b/elements/contextmenu.js
index 9776ad8..515d0b6 100644
--- a/elements/contextmenu.js
+++ b/elements/contextmenu.js
@@ -120,7 +120,7 @@ let DuoyunContextmenuElement = (() => {
         static async open(contextmenu, options = {}) {
             if (Array.isArray(contextmenu) && contextmenu.length === 0)
                 throw new Error('menu length is 0');
-            const { activeElement, openLeft, x = 0, y = 0, maskClosable = true } = options;
+            const { activeElement, openLeft, x, y, maskClosable = true } = options;
             const menuObject = getMenuObject(contextmenu);
             const menu = menuObject.menu;
             const header = menuObject.header || options.header;
@@ -128,12 +128,23 @@ let DuoyunContextmenuElement = (() => {
             const maxHeight = menuObject.maxHeight || options.maxHeight;
             const searchable = menuObject.searchable || options.searchable;
             toggleActiveState(activeElement, true);
+            const prevMenu = contextmenuStore.menuStack.at(0);
             contextmenuStore({
                 activeElement,
-                onlyActive: !!x || !!y,
+                onlyActive: x !== undefined || y !== undefined,
                 openLeft,
                 maskClosable,
-                menuStack: [{ x, y, menu, searchable, header, width, maxHeight }],
+                menuStack: [
+                    {
+                        x: x ?? prevMenu?.x ?? 0,
+                        y: y ?? prevMenu?.y ?? 0,
+                        menu,
+                        searchable,
+                        header,
+                        width,
+                        maxHeight,
+                    },
+                ],
             });
             if (!ContextMenu.instance) {
                 const ele = new ContextMenu();
