diff --git a/elements/avatar.js b/elements/avatar.js
index ffc5ba97fc393886b6420acb80b416338abad8d8..71670509c3c8ce121c810b58ded0d466c3686ec9 100644
--- a/elements/avatar.js
+++ b/elements/avatar.js
@@ -61,13 +61,13 @@ const style = css `
   :host([size='large']) {
     width: 3.8em;
   }
-  .slot:has-slotted {
+  .slot {
     place-items: center;
     place-content: center;
     border-color: ${theme.textColor};
   }
   .img,
-  .slot:has-slotted {
+  .slot {
     display: flex;
     width: 100%;
     aspect-ratio: 1;
diff --git a/elements/sort-box.js b/elements/sort-box.js
index 15a623e93a3366c2baf72a815dec34a0b9979923..e66e03d79efd6d6dc62c8e59c1883e0a19ed6879 100644
--- a/elements/sort-box.js
+++ b/elements/sort-box.js
@@ -130,7 +130,7 @@ let DuoyunSortItemElement = (() => {
     let _classDescriptor;
     let _classExtraInitializers = [];
     let _classThis;
-    let _classSuper = DuoyunGestureElement;
+    let _classSuper = GemElement;
     let _handle_decorators;
     let _handle_initializers = [];
     let _handle_extraInitializers = [];
